mm2src/coins_activation/src/eth_with_token_activation.rs

```diff
@@ -0,0 +1,233 @@
+    type ActivationRequest = EthWithTokensActivationRequest;
+    type PlatformProtocolInfo = CoinProtocol;
+    type ActivationResult = EthWithTokensActivationResult;
+    type ActivationError = EthActivationV2Error;
+
+    async fn enable_platform_coin(
+        ctx: MmArc,
+        ticker: String,
+        platform_conf: Json,
+        activation_request: Self::ActivationRequest,
+        protocol: Self::PlatformProtocolInfo,
+        priv_key: &[u8],
+    ) -> Result<Self, MmError<Self::ActivationError>> {
+        let platform_coin = eth_coin_from_conf_and_request_v2(
+            &ctx,
+            &ticker,
+            &platform_conf,
+            activation_request.platform_request,
+            priv_key,
+            protocol,
+        )
+        .await?;
+
+        Ok(platform_coin)
+    }
+
+    fn token_initializers(
+        &self,
+    ) -> Vec<Box<dyn TokenAsMmCoinInitializer<PlatformCoin = Self, ActivationRequest = Self::ActivationRequest>>> {
+        vec![Box::new(Erc20Initializer {
+            platform_coin: self.clone(),
+        })]
+    }
+
+    async fn get_activation_result(&self) -> Result<EthWithTokensActivationResult, MmError<EthActivationV2Error>> {
+        let my_address = self.my_address().map_err(EthActivationV2Error::InternalError)?;
+        let pubkey = self
+            .get_public_key()
+            .map_err(|e| EthActivationV2Error::InternalError(e.to_string()))?;
+
+        let current_block = self
+            .current_block()
+            .compat()
+            .await
+            .map_err(EthActivationV2Error::InternalError)?;
+
+        let eth_balance = self
+            .my_balance()
+            .compat()
+            .await
+            .map_err(|e| EthActivationV2Error::CouldNotFetchBalance(e.to_string()))?;
+        let token_balances = self
+            .get_tokens_balance_list()
+            .await
+            .map_err(|e| EthActivationV2Error::CouldNotFetchBalance(e.to_string()))?;
+
+        let mut result = EthWithTokensActivationResult {
+            current_block,
+            eth_addresses_infos: HashMap::new(),
+            erc20_addresses_infos: HashMap::new(),
+        };
+
+        result
+            .eth_addresses_infos
+            .insert(my_address.to_string(), CoinAddressInfo {
+                derivation_method: DerivationMethod::Iguana,
+                pubkey: pubkey.clone(),
+                balances: eth_balance,
+            });
+
+        result
+            .erc20_addresses_infos
+            .insert(my_address.to_string(), CoinAddressInfo {
+                derivation_method: DerivationMethod::Iguana,
+                pubkey,
+                balances: token_balances,
+            });
+
+        Ok(result)
+    }
+
+    fn start_history_background_fetching(
+        &self,
+        _metrics: MetricsArc,
+        _storage: impl TxHistoryStorage + Send + 'static,
+        _initial_balance: BigDecimal,
+    ) -> AbortHandle {
+        todo!()
```

Comment:
Someone definitely can try to pass `tx_history: true`
https://github.com/KomodoPlatform/atomicDEX-API/blob/ae790660b0fc7350c7132bc7f0003454a8af4ea3/mm2src/coins_activation/src/platform_coin_with_tokens.rs#L317-L318

Reply:
Updated with printing error message